zcash_primitives/src/transaction/builder.rs

```diff
@@ -63,11 +60,19 @@ pub enum Error {
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
-            Error::ChangeIsNegative(amount) => {
-                write!(f, "Change is negative ({:?} zatoshis)", amount)
+            Error::BalanceFailure(amount) => {
```

Comment:
Please document the meaning of this value's sign, both in the doc for `Error::BalanceFailure`, and the error message below.